app/scripts/analysis/tools/powercalc/PowercalcMainCtrl.ctrl.coffee

```diff
@@ -0,0 +1,1529 @@
+    if name is "TwoTGUI_mean2"
+      @TwoTGUI_mean2 = parseFloat(val)
+    if name is "TwoTGUI_sigma1"
+      @TwoTGUI_sigma1 = parseFloat(val)
+    if name is "TwoTGUI_sigma2"
+      @TwoTGUI_sigma2 = parseFloat(val)
+    if name is "TwoTGUI_power"
+      @TwoTGUI_power = parseFloat(val)
+    if key is 13
+      if name is "TwoTGUI_power"
+        @TwoTGUI_powerTon()
+        return
+      @TwoTGUI_update()
+      return
+  TwoTGUI_show_help: () ->
+    if @TwoTGUI_help is true
+      $('#TwoTGUI_H').val "Show Help"
+    else
+      $('#TwoTGUI_H').val "Hide Help"
+    @TwoTGUI_help = !@TwoTGUI_help
+    return
+  TwoTGUI_t_test: () ->
+    v1 = Math.pow(@TwoTGUI_sigma1,2) / @TwoTGUI_n1
+    v2 = Math.pow(@TwoTGUI_sigma2,2) / @TwoTGUI_n2
+    df =  Math.pow((v1 + v2),2) / (Math.pow(v1,2) / (@TwoTGUI_n1 - 1.0) + Math.pow(v2,2) / (@TwoTGUI_n2 - 1.0))
+    @TwoTGUI_t = @tdistr(df, 1-@TwoTGUI_alpha)
+    @TwoTGUI_pvalue = @tprob(df, @TwoTGUI_t)
+  TwoTGUI_reset: () ->
+    @TwoTGUI_n1 = 10
+    @TwoTGUI_n2 = 10
+    @TwoTGUI_maxn1 = 20
+    @TwoTGUI_maxn2 = 20
+    @TwoTGUI_maxn = 20
+    @TwoTGUI_mean1 = 10
+    @TwoTGUI_mean2 = 10
+    @TwoTGUI_meanMax1 = 20
+    @TwoTGUI_meanMax2 = 20
+    @TwoTGUI_meanMax = 20
+    @TwoTGUI_sigma1 = 20
+    @TwoTGUI_sigma2 = 20
+    @TwoTGUI_sigmaMax1 = 40
+    @TwoTGUI_sigmaMax2 = 40
+    @TwoTGUI_sigmaMax = 40
+    @TwoTGUI_alpha = 0.010
+    @TwoTGUI_power = 0
+    @TwoTGUI_powerMax = 1 
+    @TwoTGUI_t = 0
+    @TwoTGUI_pvalue = 0
+    @TwoTGUI_mode = "Two Tailed"
+    @TwoTGUI_update()
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+  tprob: ($n, $x) ->
+    if $n <= 0
+      throw 'Invalid n: $n\n'
+      ### degree of freedom ###
+    @precision_string @_subtprob($n - 0, $x - 0)
+
+  integer: ($i) ->
+    if $i > 0
+      Math.floor $i
+    else
+      Math.ceil $i
+
+  precision_string: ($x) ->
+    if $x
+      @round_to_precision $x, @precision($x)
+    else
+      '0'
+
+  round_to_precision: ($x, $p) ->
+    $x = $x * 10 ** $p
+    $x = Math.round($x)
+    $x / 10 ** $p
+
+  precision: ($x) ->
+    Math.abs @integer(@log10(Math.abs($x)) - @SIGNIFICANT)
+
+  _subtprob: ($n, $x) ->
+    $a = undefined
+    $b = undefined
+    $w = Math.atan2($x / Math.sqrt($n), 1)
+    $z = Math.cos($w) ** 2
+    $y = 1
+    $i = $n - 2
+    while $i >= 2
+      $y = 1 + ($i - 1) / $i * $z * $y
+      $i -= 2
+    if $n % 2 == 0
+      $a = Math.sin($w) / 2
+      $b = .5
+    else
+      $a = if $n == 1 then 0 else Math.sin($w) * Math.cos($w) / Math.PI
+      $b = .5 + $w / Math.PI
+    @max 0, 1 - $b - ($a * $y)
+
+  log10: ($n) ->
+    Math.log($n) / Math.log(10)
+
+  max: () ->
+    $max = arguments[0]
+    $i = 0
+    while $i < arguments.length
+      if $max < arguments[$i]
+        $max = arguments[$i]
+      $i++
+    $max
+
+  tdistr: ($n, $p) ->
+    if $n <= 0
+      throw 'Invalid n: $n\n'
+    if $p <= 0 or $p >= 1
+      throw 'Invalid p: $p\n'
+    @precision_string @_subt($n - 0, $p - 0)
+
+  _subt: ($n, $p) ->
```

Comment:
also should rename this functions to correspond our naming, i.e. remove underscores and make them camelCase